### PR TITLE
Fix missing data in email submissions

### PR DESCRIPTION
**Pull request by aristath**
## What?
Fixes an issue with form submissions from the experimental form block, when using the email method.

## Why?
Because emails were not being sent when submitting a form

## How?
Adds missing data to the script, which then lets the form submission email to be properly sent.

## Testing Instructions
1. Add a form block in a post/page
2. In the form block's settings, select "Send an email" in the "submissions method" firld, and then enter your email address in the "email for form submissions" field
3. Save the post and view in on the front
4. Submit something in the form, and you should now get the email (Note: this will also depend on #55690. If that PR doesn't get merged before this one, then you'll need to manually change the type of the button to `submit` from your browser's tools, so the form can be properly submitted)

### Testing Instructions for Keyboard
Not applicable

### CODE REVIEW

**Review by draganescu — COMMENTED** [review]
I'd love to know what was wrong before, in the code - I understand the email was not sent, why?. The code changes are not obvious to me :P)

**Comment by aristath** [comment]
> I'd love to know what was wrong before, in the code - I understand the email was not sent, why?. The code changes are not obvious to me :P)

email forms have `<form action="mailto:name@example.com">`. If the user has JS enabled on the browser, then a script on the frontend then takes care of making an AJAX request to send the email (default behaviour in browsers would be to open their email client with the body containing what they entered in the form).
That script was missing some data, and the `block_core_form_send_email` was unable to send the email. Specifically, the most important bug was that the `mailto:` action was not passed-on to in the AJAX request, and `wp_mail` didn't know _where_ to send the email 😅

File: packages/block-library/src/form/view.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ document.querySelectorAll( 'form.wp-block-form' ).forEach( function ( form ) {
 		formData.formAction = form.action;
 		formData._ajax_nonce = wpBlockFormSettings.nonce;
 		formData.action = wpBlockFormSettings.action;
+		formData._wp_http_referer = window.location.href;
```
Comment (draganescu) [inline packages/block-library/src/form/view.js:23]: Overriding `_wp_http_referer` like this makes me anxious. Why is this needed?

File: packages/block-library/src/form/index.php [hunk 2]
```diff
@@ -109,7 +109,7 @@ function block_core_form_send_email() {
 
 	// Send the email.
 	$result = wp_mail(
-		str_replace( 'mailto:', '', $params['wp-email-address'] ),
+		str_replace( 'mailto:', '', $params['formAction'] ),
```
Comment (draganescu) [inline packages/block-library/src/form/index.php:112]: This seems like the main bug fixed here, right?

File: packages/block-library/src/form/index.php [hunk 3]
```diff
@@ -96,7 +96,7 @@ function block_core_form_send_email() {
 		'<a href="' . esc_url( get_site_url( null, $params['_wp_http_referer'] ) ) . '">' . get_bloginfo( 'name' ) . '</a>'
 	);
 
-	$skip_fields = array( 'formAction', '_ajax_nonce', 'action' );
+	$skip_fields = array( 'formAction', '_ajax_nonce', 'action', '_wp_http_referer' );
```
Comment (draganescu) [inline packages/block-library/src/form/index.php:99]: Maybe too late but it'd be great if we could instead of keeping alist of fields to skip, find a way to identify what to include - like if we had a prefix for all fields that should be emailed or something?

**Review by draganescu — COMMENTED** [review]
I understand the explanation, left some comments around some code changes.

File: packages/block-library/src/form/view.js (hunk 1 above)
Comment (aristath) [reply packages/block-library/src/form/view.js:23]: The data is not actually sent with the AJAX request... so we're just adding it here.

File: packages/block-library/src/form/index.php (hunk 2 above)
Comment (aristath) [reply packages/block-library/src/form/index.php:112]: Correct

File: packages/block-library/src/form/index.php (hunk 3 above)
Comment (aristath) [reply packages/block-library/src/form/index.php:99]: Well, the args that need to be ignored are finite and known... That's why I thought it best to keep it clean and not add prefixes to the other ones. 😃

Comment (draganescu) [reply packages/block-library/src/form/index.php:99]: > the args that need to be ignored are finite and known

Today! But one day some hidden thing _don't_email_me will show up and be emailed :)

**Comment by paaljoachim** [comment]
I am looking forward to getting this one merged. As I will then be able to test out the Forms block.

**Review by draganescu — APPROVED** [review]
Let's get this in as is to enable this functionality now in the experiment.

**Comment by ingereck** [comment]
Is there a fix for the Chrome warning: "This form is not secure. Auto-fill has been turned off" (Mixed content: ...`mailto:info@ingereck.net`)?
